libmscore/repeatlist.cpp

```diff
@@ -298,121 +346,159 @@ void RepeatList::unwind()
       dump();
       }
 
+//---------------------------------------------------------
+//   preProcessVoltas
+//    determine the real effective endpoint of voltas
+//    because an open volta doesn't necessarily end on its end anchorpoint
+//    MUST be called before unwinding(Section) logic!
+//---------------------------------------------------------
+
+void RepeatList::preProcessVoltas()
+      {
+      Volta* nextVolta = nullptr; //work backwards as a volta can end because the next one starts
+      for (auto rit = _score->spanner().crbegin(); rit != _score->spanner().crend(); ++rit)
+            {
+            Spanner* s = (*rit).second;
+            if (s->type() != Element::Type::VOLTA)
+                  continue;
+            Volta* volta = static_cast<Volta*>(s);
+            if (volta) {
```

Comment:
I would reduce number of internal if scopes like
`if (!volta) {`
`  return;`
`  }`
and then continue without extra if scope. This approach seriously simplifies reading code.

Also,
`if (_voltaRanges.find(volta) != _voltaRanges.end()) {`
`qDebug("Found same volta twice in the spannermap");`
`continue;`
`}`
and following code without extra tab as well.